modules/build/src/main/scala/scala/build/bsp/BspImpl.scala

```diff
@@ -578,8 +573,11 @@ final class BspImpl(
           }
         }
         val newInputs      = value(argsToInputs(ideInputs.args))
+        val newHash        = newInputs.sourceHash()
         val previousInputs = currentBloopSession.inputs
-        if (newInputs == previousInputs) CompletableFuture.completedFuture(new Object)
+        val previousHash   = currentBloopSession.inputsHash
+        if newInputs == previousInputs && newHash == previousHash then
```

Comment:
```suggestion
        if newHash == previousHash then
```

It should be sufficient, as the sourceHash method calculates a hash that includes [the file path](https://github.com/VirtusLab/scala-cli/blob/5264acdea6de5509fc47b291456607c704eb7792/modules/build/src/main/scala/scala/build/input/Inputs.scala#L104).

Reply:
I think it may not be sufficient in some edge cases, as Inputs has some fields beyond the actual elements (like i.e. if Markdown sources are to be included). I'd leave it as it is, just to be safe, the extra comparison is a cheap enough operation